websockets/handlers/simple_handshake_wsh.py

```diff
@@ -1,13 +1,25 @@
 #!/usr/bin/python
 
-from mod_pywebsocket import common, msgutil, util
-from mod_pywebsocket.handshake import hybi
+import struct
+
+from mod_pywebsocket import common, stream
+from mod_pywebsocket.handshake import AbortedByUserException, hybi
 
 
 def web_socket_do_extra_handshake(request):
+    # Send simple response header. This test implements the handshake
+    # manually. It's not clear why.
     request.connection.write('HTTP/1.1 101 Switching Protocols:\x0D\x0AConnection: Upgrade\x0D\x0AUpgrade: WebSocket\x0D\x0ASet-Cookie: ws_test=test\x0D\x0ASec-WebSocket-Origin: '+request.ws_origin+'\x0D\x0ASec-WebSocket-Accept: '+hybi.compute_accept(request.headers_in.get(common.SEC_WEBSOCKET_KEY_HEADER))[0]+'\x0D\x0A\x0D\x0A')
-    return
+    # Send a clean close frame.
+    body = stream.create_closing_handshake_body(1000, '')
+    request.connection.write(stream.create_close_frame(body));
+    # Wait for the responding close frame from the user agent. It's not possible
+    # to use the stream methods at this point because the stream hasn't been
+    # established from pywebsocket's point of view. Instead just read the
+    # appropriate number of bytes and assume they are correct.
+    request.connection.read(8)
```

Comment:
Don't we need to emulate _stream_base.receive_bytes() here?

Reply:
I think it's safe to assume that if we got any bytes at all then the close frame has been sent.

It's possible that a browser could send the close frame fragmented into two parts, the server closes the connection after the first fragment and the browser gets a "Connection reset by peer" error and sets wasClean to false. But I know of no reason why a browser would fragment the close frame.

This handler is already complicated and probably pointless, so I don't want to complicate it any further.

Reply:
ok